presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestGroupInnerJoinsByConnectorRuleSet.java

```diff
@@ -0,0 +1,490 @@
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+public class TestGroupInnerJoinsByConnectorRuleSet
+{
+    public static final String CATALOG_SUPPORTING_JOIN_PUSHDOWN = "catalog_join_pushdown_supported";
+    public static final String LOCAL = "local";
+    public static final String TEST_SCHEMA = "test-schema";
+    public static final String TEST_TABLE = "test-table";
+    public static final String OTHER_CATALOG_SUPPORTING_JOIN_PUSHDOWN = "other_catalog_join_pushdown_supported";
+    private PlanBuilder planBuilder;
+    private RuleTester tester;
+
+    @BeforeClass
+    public void setUp()
+    {
+        LocalQueryRunner runner = new LocalQueryRunner(TEST_SESSION);
+        ConnectorFactory pushdownConnectorFactory = new TestingJoinPushdownConnectorFactory()
+        {
+            @Override
+            public String getName()
+            {
+                return "tpch_with_join_pushdown";
+            }
+        };
+        ConnectorFactory pushdownConnectorFactory1 = new TestingJoinPushdownConnectorFactory()
+        {
+            @Override
+            public String getName()
+            {
+                return "tpch_with_join_pushdown1";
+            }
+        };
+        runner.createCatalog(CATALOG_SUPPORTING_JOIN_PUSHDOWN, pushdownConnectorFactory, ImmutableMap.of());
+        runner.createCatalog(OTHER_CATALOG_SUPPORTING_JOIN_PUSHDOWN, pushdownConnectorFactory1, ImmutableMap.of());
+
+        tester = new RuleTester(
+                ImmutableList.of(),
+                RuleTester.getSession(ImmutableMap.of(INNER_JOIN_PUSHDOWN_ENABLED, "true", INEQUALITY_JOIN_PUSHDOWN_ENABLED, "true"), createTestingSessionPropertyManager()),
+                runner,
+                new TpchConnectorFactory(1));
+
+        planBuilder = new PlanBuilder(TEST_SESSION, new PlanNodeIdAllocator(), runner.getMetadata());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        closeAllRuntimeException(tester);
+        tester = null;
+    }
+
+    @Test
+    public void testDoesNotPushDownOuterJoin()
+    {
+        String connectorName = "test_catalog";
+
+        VariableReferenceExpression left = newBigintVariable("a1");
+        VariableReferenceExpression right = newBigintVariable("a2");
+        EquiJoinClause joinClause = new EquiJoinClause(left, right);
+
+        assertGroupInnerJoinsByConnectorRuleSet().on(p -> p.join(FULL,
+                tableScan(connectorName, "a1", "b1"),
+                tableScan(connectorName, "a2", "b2"),
+                joinClause)).doesNotFire();
+    }
+
+    @Test
+    public void testDoesNotPushDownTwoDifferentConnectors()
```

Comment:
Let's enhance this test so oin 3 tables `(t1 IJ t2) IJ t3`, and `t1` and `t3` are from a catalog that can be joined, while `t2` is from a different catalog